control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -19,39 +19,305 @@ package broker
+		// For now, we use the configuration map only on the dispatcher side, which means we don't lose availability,
+		// since the receiver will accept events. So, log out the error and move on to the next step.
+		logger.Warn(
+			"failed to update dispatcher pod annotation to trigger an immediate config map refresh",
+			zap.Error(err),
+		)
+
+		statusConditionManager.failedToUpdateDispatcherPodsAnnotation(broker, err)
+	} else {
+		logger.Debug("updated dispatcher pod annotation")
+	}
+
+	return statusConditionManager.reconciled()
 }
 
 func (r *Reconciler) FinalizeKind(ctx context.Context, broker *eventing.Broker) reconciler.Event {
-	r.logger.Debug("finalizing Broker", zap.Any("broker", broker))
+	logger := logger(ctx, broker)
+
+	logger.Debug("finalizing broker")
+
+	// Get brokers and triggers config map.
+	brokersTriggersConfigMap, err := r.GetBrokersTriggersConfigMap()
+	if err != nil {
+		return fmt.Errorf("failed to get brokers and triggers config map %s: %w", r.Configs.BrokersTriggersConfigMapAsString(), err)
+	}
+
+	logger.Debug("got brokers and triggers config map")
+
+	// Get brokersTriggers data.
+	brokersTriggers, err := r.GetBrokersTriggers(logger, brokersTriggersConfigMap)
+	if err != nil {
+		return fmt.Errorf("failed to get brokers and triggers: %w", err)
+	}
+
+	logger.Debug(
+		"got brokers and triggers data from config map",
+		zap.Any(base.BrokersTriggersDataLogKey, log.BrokersMarshaller{Brokers: brokersTriggers}),
+	)
+
+	brokerIndex := findBroker(brokersTriggers, broker)
+	if brokerIndex == noBroker {
+		return fmt.Errorf("broker %s/%s not found in config map", broker.Namespace, broker.Name)
+	}
+	deleteBroker(brokersTriggers, brokerIndex)
+
+	logger.Debug("broker deleted", zap.Int("index", brokerIndex))
+
+	// Update the configuration map with the new brokersTriggers data.
+	if err := r.UpdateBrokersTriggersConfigMap(brokersTriggers, brokersTriggersConfigMap); err != nil {
+		return fmt.Errorf("failed to update configuration map: %w", err)
+	}
+
+	logger.Debug("brokers and triggers config map updated")
+
+	// There is no need to update volume generation and dispatcher pod annotation, updates to the config map will
+	// eventually be seen by the dispatcher pod and resources will be deleted accordingly.
+
+	topic, err := r.deleteTopic(broker)
+	if err != nil {
+		return fmt.Errorf("failed to delete topic %s: %w", topic, err)
+	}
+
+	logger.Debug("topic deleted", zap.String("topic", topic))
 
 	return reconciledNormal(broker.Namespace, broker.Name)
 }
 
-func reconciledNormal(namespace, name string) reconciler.Event {
-	return reconciler.NewEvent(
-		corev1.EventTypeNormal,
-		brokerReconciled,
-		fmt.Sprintf(`%s reconciled: "%s/%s"`, broker, namespace, name),
-	)
+func incrementVolumeGeneration(generation uint64) uint64 {
+	return (generation + 1) % (math.MaxUint64 - 1)
+}
+
+func (r *Reconciler) createTopic(broker *eventing.Broker) (string, error) {
```

Comment:
Can you move the functions to manage topics in another file? it may help a bit the readability